apiserver/registration.go

```diff
@@ -233,3 +208,10 @@ func (h *registerUserHandler) getSecretKeyLoginResponsePayload(
 	}
 	return &payload, nil
 }
+
+// ConnectorEnviron defines a mock environ type that can be used as the return
+// value from the apiserver's provider factory.
+type ConnectorEnviron interface {
```

Comment:
sorry, I thought the comment explained it. We need to mock the provider factory for testing. The return value from the provider factory has to implement `providertracker.Provider`. Previously we were using just a mock of `environs.ConnectorInfo`. However, this could not be set as the return value of the fake provider factory, as it doesn't implement `providertracker.Provider`. Hence I defined the new type so we could generate a mock from it.

Reply:
Yeah have a look where it is though. We don't want to compile this type into Juju. You can still just mock the core interface needed and use composition with a nil provider to get the same result as well. This does not need to live outside of the testing code to mock it.

Reply:
done